### ISSUE TITLE
Is there an implementation of MobileNetv3-YOLOv5s, ShuffleNetv2-YOLOv5s?

### ISSUE THREAD

**Issue by River-Cold** [AlanLi1997/slim-neck-by-gsconv#19]
![image](https://user-images.githubusercontent.com/80696033/212290961-f4ce388c-6bf1-4e24-b771-72296259055a.png)
When I reproduced the backbone network comparison experiment in the table 9 of section 4.3 of the paper, I found that the number of parameters and computation of ShuffleNetv2 network are much less than GhostNet, but the number of parameters and computation in the table of your paper are more than GhostNet, could you please give the yaml implementation  of MobileNetv3-YOLOv5s, ShuffleNetv2- YOLOv5s?

**Comment by AlanLi1997** [comment]
> ![image](https://user-images.githubusercontent.com/80696033/212290961-f4ce388c-6bf1-4e24-b771-72296259055a.png) When I reproduced the backbone network comparison experiment in the table 9 of section 4.3 of the paper, I found that the number of parameters and computation of ShuffleNetv2 network are much less than GhostNet, but the number of parameters and computation in the table of your paper are more than GhostNet, could you please give the yaml implementation of MobileNetv3-YOLOv5s, ShuffleNetv2- YOLOv5s?

In all honesty I forgot the configuration of the original, but I found the current version in my codebase that you can try to use for reproduction.
NOTE: you may need to change the depth_multiple and width_multiple to match the original version in the paper

the MobileNetv3-YOLOv5s yaml:

------

      # parameters
      nc: 80  # number of classes
      depth_multiple: 1.0  # dont change this otherwise InvertedResidual1 will be affected
      width_multiple: 1.0  # dont change this otherwise InvertedResidual1 will be affected
      # anchors
      anchors:
        - [10,13, 16,30, 33,23]  # P3/8
        - [30,61, 62,45, 59,119]  # P4/16
        - [116,90, 156,198, 373,326]  # P5/32
      # YOLOv5 backbone
      backbone:
        # MobileNetV3-small
        # [from, number, module, args]
        [[-1, 1, Conv, [16, 2]],                         # 0-p1/2
         [-1, 1, InvertedResidual, [16,  16, 3, 2, 1, 0]],  # 1-p2/4
         [-1, 1, InvertedResidual, [24,  72, 3, 2, 0, 0]],  # 2-p3/8
         [-1, 1, InvertedResidual, [24,  88, 3, 1, 0, 0]],  # 3-p3/8
         [-1, 1, InvertedResidual, [40,  96, 5, 2, 1, 1]],  # 4-p4/16
         [-1, 1, InvertedResidual, [40, 240, 5, 1, 1, 1]],  # 5-p4/16
         [-1, 1, InvertedResidual, [40, 240, 5, 1, 1, 1]],  # 6-p4/16
         [-1, 1, InvertedResidual, [48, 120, 5, 1, 1, 1]],  # 7-p4/16
         [-1, 1, InvertedResidual, [48, 144, 5, 1, 1, 1]],  # 8-p4/16
         [-1, 1, InvertedResidual, [96, 288, 5, 2, 1, 1]],  # 9-p5/32
         [-1, 1, InvertedResidual, [96, 576, 5, 1, 1, 1]],  # 10-p5/32
         [-1, 1, InvertedResidual, [96, 576, 5, 1, 1, 1]],  # 11-p5/32
         [-1, 1, SPPF, [96, 5]],
        ]
      # YOLOv5 head
      head:
        [[-1, 1, Conv, [256, 1, 1]],
         [-1, 1, nn.Upsample, [None, 2, 'nearest']],
         [[-1, 8], 1, Concat, [1]],  # cat backbone P4
         [-1, 1, C3, [256, False]],  # 15
      
         [-1, 1, Conv, [128, 1, 1]],
         [-1, 1, nn.Upsample, [None, 2, 'nearest']],
         [[-1, 3], 1, Concat, [1]],  # cat backbone P3
         [-1, 1, C3, [128, False]],  # 19 (P3/8-small)
      
         [-1, 1, Conv, [128, 3, 2]],
         [[-1, 17], 1, Concat, [1]],  # cat head P4
         [-1, 1, C3, [256, False]],  # 22 (P4/16-medium)
      
         [-1, 1, Conv, [256, 3, 2]],
         [[-1, 13], 1, Concat, [1]],  # cat head P5
         [-1, 1, C3, [512, False]],  # 25 (P5/32-large)
      
         [[20, 23, 26], 1, Detect, [nc, anchors]],  # Detect(P3, P4, P5)
        ]
      InvertedResidual1 code:
      class InvertedResidual(nn.Module):
          def __init__(self, inp, oup, hidden_dim, kernel_size, stride, use_se, use_hs):
              super(InvertedResidual, self).__init__()
              assert stride in [1, 2]
              self.identity = stride == 1 and inp == oup
              if inp == hidden_dim:
                  self.conv = nn.Sequential(
                      # dw
                      nn.Conv2d(hidden_dim, hidden_dim, kernel_size, stride, (kernel_size - 1) // 2, groups=hidden_dim,
                                bias=False),
                      nn.BatchNorm2d(hidden_dim),
                      h_swish() if use_hs else nn.ReLU(inplace=True),
                      # Squeeze-and-Excite
                      SELayer(hidden_dim) if use_se else nn.Sequential(),
                      # pw-linear
                      nn.Conv2d(hidden_dim, oup, 1, 1, 0, bias=False),
                      nn.BatchNorm2d(oup),
                  )
              else:
                  self.conv = nn.Sequential(
                      # pw
                      nn.Conv2d(inp, hidden_dim, 1, 1, 0, bias=False),
                      nn.BatchNorm2d(hidden_dim),
                      h_swish() if use_hs else nn.ReLU(inplace=True),
                      # dw
                      nn.Conv2d(hidden_dim, hidden_dim, kernel_size, stride, (kernel_size - 1) // 2, groups=hidden_dim,
                                bias=False),
                      nn.BatchNorm2d(hidden_dim),
                      # Squeeze-and-Excite
                      SELayer(hidden_dim) if use_se else nn.Sequential(),
                      h_swish() if use_hs else nn.ReLU(inplace=True),
                      # pw-linear
                      nn.Conv2d(hidden_dim, oup, 1, 1, 0, bias=False),
                      nn.BatchNorm2d(oup),
                  )
          def forward(self, x):
              y = self.conv(x)
              if self.identity:
                  return x + y
              else:
                  return y


theShuffleNetv2- YOLOv5s yaml:

-----

      # Parameters
      nc: 80  # number of classes
      depth_multiple: 1 
      width_multiple: 0.5  
      anchors:
        - [10,13, 16,30, 33,23]  # P3/8
        - [30,61, 62,45, 59,119]  # P4/16
        - [116,90, 156,198, 373,326]  # P5/32
      # YOLOv5 v6.0 backbone
      backbone:
        # [from, number, module, args]
        [[-1, 1, Conv, [64, 6, 2, 2]],  # 0-P1/2
         [-1, 1, SV2, [128, 3, 2]],  # 1-P2/4
         [-1, 3, SV2, [128, 3, 1]],
         [-1, 1, SV2, [256, 3, 2]],
         [-1, 7, SV2, [256, 3, 1]],
      #   [-1, 1, ECA, [3]],           # 5
         [-1, 1, SV2, [512, 3, 2]],
         [-1, 3, SV2, [512, 3, 1]],
      #   [-1, 1, ECA, [3]],
         [-1, 1, SPPF, [512, 5]],  # 9
      #   [-1, 1, ECA, [3]],
        ]
      # YOLOv5 v6.0 head
      head:
        [[-1, 1, Conv, [512, 1, 1]],
         [-1, 1, nn.Upsample, [None, 1, 'nearest']],
         [[-1, 6], 1, Concat, [1]],  # cat backbone P4
         [-1, 3, C3, [512, False]],  # 14
         [-1, 1, Conv, [256, 1, 1]],
         [-1, 1, nn.Upsample, [None, 2, 'nearest']],
         [[-1, 4], 1, Concat, [1]],  # cat backbone P3
         [-1, 3, C3, [256, False]],  # 18 (P3/8-small)
         [-1, 1, Conv, [256, 3, 2]],
         [[-1, 12], 1, Concat, [1]],  # cat head P4
         [-1, 3, C3, [512, False]],  # 21 (P4/16-medium)
         [-1, 1, Conv, [512, 3, 2]],
         [8, 1, nn.MaxPool2d, [2, 2]],
         [[-2, -1], 1, Concat, [1]],  # cat head P5
         [-1, 3, C3, [1024, False]],  # 24 (P5/32-large)
         [[15, 18, 22], 1, Detect, [nc, anchors]],  # Detect(P3, P4, P5)
        ]
      SV2 code:
      class SV2(nn.Module):
             def __init__(self, inp, oup, ksize, stride, p=None):
              super(Sblock, self).__init__()
              assert stride in [1, 2]
              mid_channels = oup // 2
              pad = autopad(ksize, p)
              outputs = oup // 2 #- inp
              self.stride = stride
              self.branch_main = nn.Sequential(
                  # pw
                  nn.Conv2d(inp, mid_channels, 1, 1, 0, bias=False),
                  nn.BatchNorm2d(mid_channels),
                  nn.Hardswish(inplace=True),
                  # dw
                  nn.Conv2d(mid_channels, mid_channels, ksize, stride, pad, groups=mid_channels, bias=False),
                  nn.BatchNorm2d(mid_channels),
                  # pw-linear
                  nn.Conv2d(mid_channels, outputs, 1, 1, 0, bias=False),
                  nn.BatchNorm2d(outputs),
                  nn.Hardswish(inplace=True),
              )
              self.branch_proj = nn.Sequential(
                  # dw
                  nn.Conv2d(inp, inp, ksize, stride, pad, groups=inp, bias=False),
                  nn.BatchNorm2d(inp),
                  # pw-linear
                  nn.Conv2d(inp, inp, 1, 1, 0, bias=False),
                  nn.BatchNorm2d(inp),
                  nn.Hardswish(inplace=True),
              )
          def forward(self, old_x):
              if self.stride == 1:
                  return channel_shuffle(old_x)
              else:
                  x_proj = self.branch_proj(old_x)
                  x = self.branch_main(old_x)
                  return torch.cat((x_proj, x), 1)
